File: Java/Player.java
/**
 * プレイヤーを表すクラスです。
 * @param initialLife ゲーム開始時点の体力です
 * @param name ゲームキャラクターの名前です
 */
class Player extends GameCharacter {
  Player(int initialLife, String name) {
    super(initialLife, name);
  }

  /**
   * 行動を選択するメソッドです。 
   * プレイヤー用の処理なので標準入力を使った処理を実装しました。
   */
  @Override
  void choiceAction() {
    String prompt = this.generatePrompt();
    while (true) {
      standardIO.print(prompt);
      String input = standardIO.readLine();
      try {
        this.action = this.ACTIONS[Integer.parseInt(input) - 1];
        return;
      } catch (Exception error) {
        standardIO.printlnEmptyLine(1);
        standardIO.println("> 不正な入力です");
        standardIO.println("> 再入力してください");
        standardIO.printlnEmptyLine(1);
      }
    }
  }

  /**
   * 行動選択処理で必要になるプロンプトを作成して返すメソッドです。
   * @return プロンプト
   */
  private String generatePrompt() {
    StringBuilder prompt = new StringBuilder();
    for (int i = 0; i < this.ACTIONS.length; i += 1) {
      prompt.append("[" + (i + 1) + "] " + this.ACTIONS[i] + "    ");
    }
    prompt.append(": ");
    return prompt.toString();
  }
}

File: Kotlin/Player.kt
/**
 * プレイヤーを表すクラスです
 * 
 * @constructor
 * 
 * @param life プロパティlifeの初期値です
 * @param name プロパティnameの初期値です
 */
class Player(life: Int, name: String): GameCharacter(life, name) {

    /**
     * 行動を選択するメソッドです
     * 標準入力を利用して行動を選択します
     */
    override fun choiceAction() {
        val prompt = this.generatePrompt()
        while (true) {
            StandardIO.print(prompt)
            val input = StandardIO.readLineToIntOrNull()
            if (input != null && (1..this.actionList.size).contains(input)) {
                this.action = this.actionList[input - 1]
                return
            }
            StandardIO.printEmptyLine(1)
            StandardIO.println("> 不正な入力です")
            StandardIO.println("> 再入力してください")
            StandardIO.printEmptyLine(1)
        }
    }

    /**
     * 行動選択処理で必要となるプロンプトを作成して返します
     * 
     * @return プロンプトとなる文字列です
     */
    private fun generatePrompt(): String {
        var prompt = ""
        for ((i, v) in this.actionList.withIndex()) {
            prompt += "[${i + 1}] $v    "
        }
        prompt += ": "
        return prompt
    }
}